SO-6082: nested member create request errors

cmark opened the pull request:
Performing a concept create request with the following body throws `moduleId` is missing HTTP 400 Bad Request error, how ever concept moduleId should get propagated down to the nested create request.

```
{
  moduleId: <a-module-id>
  members: [
    {
      active: true,
      refsetId: '733073007',
      owlExpression: 'SubClassOf(:<pre-generated-concept-id> :900000000000443000)'
    }
  ],
  ...
}
```

Additionally: 
* improved the error message when `refsetId` field is missing
* deferred UUID generation to a later stage to speed up ID duplication check and request validation

cmark committed:
test(snomed): add test case to verify bug

cmark committed:
fix(snomed): remove invalid NotEmpty validation from moduleId field...

...in refset member create requests

ModuleId field can get its value from parent component requests.

Rename incorrect `referenceSetId` field to the execpted `refsetId`
field.
Remove UUID pre-generation to ensure that we only generate the UUID when
actually creating the refset member.

cmark committed:
chore(snomed): rename fields and method arguments that use...

...`referenceSetId` to `refsetId`

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetMemberCreateRequest.java:
```diff
@@ -39,22 +38,25 @@
 	
-	@NotEmpty
-	private String referenceSetId;
+	// Not @NotEmpty, it can be populated after the request is validated (eg. nested create requests)
```
The reason for removing &commat;`Nonnull` from `id` could be documented this way as well.

AAAlinaaa replied:
Or same for referencedComponentId?

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetMemberCreateRequest.java:
```diff
@@ -170,6 +172,18 @@ public Set<String> getRequiredComponentIds(TransactionContext context) {
+			try {
+				UUID.fromString(id);
+			} catch (IllegalArgumentException e) {
+				throw new BadRequestException("Refset Member uses an illegal non-UUID identifier '%s'.", id);
```
If the message is user-facing, we might need the entire incantation:

```suggestion
				throw new BadRequestException("Reference set member uses an illegal non-UUID identifier '%s'.", id);
```

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetMemberCreateRequestBuilder.java:
```diff
@@ -38,7 +38,7 @@ public final class SnomedRefSetMemberCreateRequestBuilder
 	private String id = UUID.randomUUID().toString();
```
We should leave the initial value of `id` on `null` here (on the request builder level) as well, so that it is consistent with the REST input class.

cmark replied:
Ah yes, thanks!

nagyo reviewed (approved):
LGTM!

apeteri reviewed (approved):
👍 with three small notes.

AAAlinaaa commented on snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/components/SnomedConceptApiTest.java:
```diff
@@ -1130,4 +1130,49 @@ public void reactivateConceptWithExistingOWLAxiom() throws Exception {
+					Json.object(
+						"active", true,
+						"refsetId", Concepts.REFSET_OWL_AXIOM,
+						"owlExpression", String.format("SubClassOf(:%s :%s)", moduleConceptId, Concepts.MODULE_ROOT)
```
Can we add some sort of explanation as to what this test case is testing? That the point is that there is no explicit module defined for the member?

cmark replied:
Done

AAAlinaaa reviewed (approved):
🍏

cmark committed:
fix(api): remove UUID auto-generation from refset member create request.

...builder as well

Add comments to request fields to explain why there is no validation
present for them at field level.

cmark committed:
test: update new test method name to reflect use case better
